app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -598,22 +598,30 @@ object AppRepository {
         }
         logging.d(LOG_TAG, "Got ${sessions.size} rows.")
 
-        val highlights = readHighlights()
-        for (highlight in highlights) {
-            logging.d(LOG_TAG, "$highlight")
-            for (session in sessions) {
-                if (session.sessionId == "${highlight.sessionId}") {
-                    session.highlight = highlight.isHighlight
-                }
+        val highlightedSessionIds = readHighlights()
+            .asSequence()
+            .filter { it.isHighlight }
+            .map { it.sessionId.toString() }
+            .toSet()
+
+        val highlightedSessions = sessions.map { session ->
+            if (session.sessionId in highlightedSessionIds) {
+                Session(session).apply { highlight = true }
+            } else {
+                session
             }
         }
 
         val alarmSessionIds = readAlarmSessionIds()
-        for (session in sessions) {
-            session.hasAlarm = session.sessionId in alarmSessionIds
+        val sessionsWithAlarms = highlightedSessions.map { session ->
+            if (session.sessionId in alarmSessionIds) {
+                Session(session).apply { hasAlarm = true }
+            } else {
+                session
+            }
         }
 
-        return sessions.toList()
+        return sessionsWithAlarms.toList()
```

Comment:
The `toList()` call is unnecessary.

Reply:
A habit to avoid exposing mutable lists. But redundant here. I will remove it. Thanks.